pkg/iac-providers/cft/v1/load-dir_test.go

```diff
@@ -0,0 +1,176 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package cftv1
+
+import (
+	"fmt"
+	"go.uber.org/zap"
+	"gopkg.in/src-d/go-git.v4"
+	gitConfig "gopkg.in/src-d/go-git.v4/config"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"os"
+	"path/filepath"
+	"syscall"
+	"testing"
+
+	"github.com/accurics/terrascan/pkg/iac-providers/output"
+	"github.com/accurics/terrascan/pkg/utils"
+	"github.com/hashicorp/go-multierror"
+)
+
+const (
+	repoURL  = "https://github.com/accurics/KaiMonkey.git"
+	branch   = "master"
+	basePath = "artifacts"
+	provider = "cft"
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	shutdown()
+	os.Exit(code)
+}
+
+func TestLoadIacDir(t *testing.T) {
+	testDataDir := "testdata"
+
+	pathErr := &os.PathError{Op: "lstat", Path: "not-there", Err: syscall.ENOENT}
+	if utils.IsWindowsPlatform() {
+		pathErr = &os.PathError{Op: "CreateFile", Path: "not-there", Err: syscall.ENOENT}
+	}
+
+	table := []struct {
+		name    string
+		dirPath string
+		cftv1   CFTV1
+		want    output.AllResourceConfigs
+		wantErr error
+	}{
+		{
+			name:    "empty config",
+			dirPath: filepath.Join(testDataDir, "testfile"),
+			cftv1:   CFTV1{},
+			wantErr: multierror.Append(fmt.Errorf("no directories found for path %s", filepath.Join(testDataDir, "testfile"))),
+		},
+		{
+			name:    "load config dir with sub directories",
+			dirPath: testDataDir,
+			cftv1:   CFTV1{},
+			wantErr: nil,
+		},
+		{
+			name:    "invalid dirPath",
+			dirPath: "not-there",
+			cftv1:   CFTV1{},
+			wantErr: multierror.Append(pathErr),
+		},
+		{
+			name:    "load valid dir",
+			dirPath: filepath.Join(testDataDir, "s3"),
+			cftv1:   CFTV1{},
+			wantErr: nil,
+		},
+	}
+
+	for _, tt := range table {
+		t.Run(tt.name, func(t *testing.T) {
+			_, gotErr := tt.cftv1.LoadIacDir(tt.dirPath, false)
+			me, ok := gotErr.(*multierror.Error)
+			if !ok {
+				t.Errorf("expected multierror.Error, got %T", gotErr)
+			}
+			if tt.wantErr == nil {
+				if err := me.ErrorOrNil(); err != nil {
+					t.Errorf("unexpected error; gotErr: '%v', wantErr: '%v'", gotErr, tt.wantErr)
+				}
+			} else if me.Error() != tt.wantErr.Error() {
+				t.Errorf("unexpected error; gotErr: '%v', wantErr: '%v'", gotErr, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestCFTMapper(t *testing.T) {
+	root := filepath.Join(basePath, provider)
+	dirList := make([]string, 0)
+	err := filepath.Walk(root, func(filePath string, fileInfo os.FileInfo, err error) error {
+		if fileInfo != nil && fileInfo.IsDir() {
+			dirList = append(dirList, filePath)
+		}
+		return err
+	})
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	cftv1 := CFTV1{}
+	for i := 1; i < len(dirList); i++ {
+		dir := dirList[i]
+		t.Run(dir, func(t *testing.T) {
+			_, gotErr := cftv1.LoadIacDir(dir, false)
+			_, ok := gotErr.(*multierror.Error)
```

Comment:
this assertion doesn't seem to be correct, we are just expecting the type of error to be multi-error. we should be asserting on expected errors (if any)